src/foam/core/AbstractFObjectArrayPropertyInfo.java

```diff
@@ -34,20 +34,26 @@ public Object fromString(String value) {
   public Object fromXML(X x, XMLStreamReader reader) {
     List objList = new ArrayList();
     String startTag = reader.getLocalName();
-
+    Class defaultClass = this.getClassInfo().getObjClass();
+    Class<?> clazz;
+    try {
+       clazz = Class.forName(this.of());
+       defaultClass = clazz;
+    } catch (Throwable th) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error("Can not crate class from class name");
+    }
     try {
       int eventType;
       while ( reader.hasNext() ) {
-        eventType = reader.next();
+        eventType = reader.getEventType();
         switch ( eventType ) {
           case XMLStreamConstants.START_ELEMENT:
             // Nested object in array
-            if ( reader.getLocalName().equals("object") ) {
-              FObject o = XMLSupport.createObj(x, reader);
+              FObject o = XMLSupport.createObj(x, reader, defaultClass);
               if ( o != null ) {
                 objList.add(o);
               }
-            }
             break;
           case XMLStreamConstants.END_ELEMENT:
```

Comment:
Here, the function doesn't know that it is parsing array. So when it hits </Stmt> it returns array with only 1 entry. For the example below the parsed object will have only last "Stmt" entry. 

I can't check reader.next() since it would move pointer and break logic for parsing. Clone function for reader is disabled.
```
<Document>
    <BkToCstmrStmt>
        <Stmt>
            <Id>9pzXREqoMo</Id>
        </Stmt>
        <Stmt>
            <Id>8pzXREqaMO</Id>
        </Stmt>
        <Stmt>
            <Id>ApzXREqaMO</Id>
        </Stmt>
    </BkToCstmrStmt>    
</Document>
```